src/main/java/org/mitre/synthea/export/BB2RIFExporter.java

```diff
@@ -957,12 +1027,22 @@ private void exportInpatient(Person person, long stopTime)
           fieldValues.put(INPATIENT.PRNCPAL_DGNS_CD, mappedDiagnosisCodes.get(0));
         }
       }
+
+      // Check for external code...
+      setExternalCode(person, fieldValues,
```

Comment:
You could move these two calls to setExternalCode inside the if block starting on line 1039 since they both rely on there being a INPATIENT.PRNCPAL_DGNS_CD set previously.

Reply:
OK, good idea.